File: Practica 4/FiguraGeometrica.java
public class FiguraGeometrica {
    
    double area(double radio) {
        return Math.PI * radio * radio;
    }
    
    double area(double base, float altura) {
        return base * altura;
    }
    
    double area(double base, double altura) { 
        return 0.5 * base * altura;
    }
    
    double area(double base1, double base2, double altura) {
        return 0.5 * (base1 + base2) * altura;
    }
    
    double area(float lado) { 
        return (3 * Math.sqrt(3) / 2) * lado * lado;
    }
    
    public static void main(String[] args) {
        FiguraGeometrica f1 = new FiguraGeometrica();
        FiguraGeometrica f2 = new FiguraGeometrica();
        FiguraGeometrica f3 = new FiguraGeometrica();
        FiguraGeometrica f4 = new FiguraGeometrica();
        FiguraGeometrica f5 = new FiguraGeometrica();
        
        System.out.println("Circulo: " + f1.area(5));
        System.out.println("Rectangulo:" + f2.area(2,3));
        System.out.println("Triangulo Rectangulo: " + f3.area(4,6));
        System.out.println("Trapecio: " + f4.area(4,6,8));
        System.out.println("Hexagono: " + f5.area(7));
    }
    
    
}

File: Practica 4/FiguraGeometrica.py
from multimethod import multimethod
import math

class FiguraGeometrica:

    @multimethod
    def area(self, radio: float):
        return math.pi * radio * radio

    @multimethod
    def area(self, base: float, altura: float):
        return base * altura

    @multimethod
    def area(self, base: float, altura: int):
        return 0.5 * base * altura

    @multimethod
    def area(self, base1: float, base2: float, altura: float):
        return 0.5 * (base1 + base2) * altura

    @multimethod
    def area(self, lado: int):
        return (3 * math.sqrt(3) / 2) * lado * lado


figura = FiguraGeometrica()
f1 = figura.area(5.0)
f2 = figura.area(2.0, 3.0)
f3 = figura.area(4.0, 6)
f4 = figura.area(4.0, 6.0, 8.0)
f5 = figura.area(7)

print("Circulo:", f1)  
print("Rectangulo:", f2) 
print("Triangulo:",f3) 
print("Trapecio:", f4)  
print("Hexagono:", f5)  